testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -236,10 +263,12 @@ impl RunLoop {
             });
         }
 
-        let mut burnchain_height = 1;
+        let mut block_height = burnchain_tip.block_snapshot.block_height;
```

Comment:
If the node is booting up, then shouldn't the first target burnchain height be equal to the first burnchain block height at which the node starts indexing?

Reply:
Not sure, I think this is what we want, however there might be an issue line 271, where `target_burnchain_block_height` should probably be:

```
        let blocks_before_reward_cycle = (burnchain_height - burnchain_config.first_block_height) % pox_constants.reward_cycle_length as u64;
        target_burnchain_block_height = burnchain_height + blocks_before_reward_cycle;
```
No? (cc @kantai)

Reply:
I think there's a little confusion here with line 216 -- `burnchain.wait_for_sortitions()`, which will try to wait until the sortition db has caught up with the burnchain tip, and then the activity of the pox sync watchdog, which also does some initial synchronization between the burnchain tip and the sortition db.

I'm not sure what the best way to clean that up is -- the pox sync watchdog should probably be in charge of handling this, but I think if we just blindly removed the `burnchain.wait_for_sortitions()` call, the neon_node would probably receive _way_ more events during start up than it currently does (the `block_height` variable is used to ignore sending "mine block" directives before the first tip).

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2073/commits/8285cb26fce46c9d21bba5786d9fa399ba615a09

Reply:
I think the code will need to do the following:

* Fetch _all_ the bitcoin headers, starting from the genesis header (Stacks 1.0 does this, for example)
* Once all bitcoin headers have been downloaded, proceed to download and process bitcoin blocks starting at `BITCOIN_TESTNET_FIRST_BLOCK_HEIGHT` (or similar -- whatever is in the `BurnchainParameters`), in batches of one reward cycle.  The PoX watchdog should be used to determine when to advance to the next reward cycle.
* Once all reward cycles have been processed, proceed to attempt to mine blocks.

To do this, I think the runloop's setup code needs to have an extra bootstrapping step that will fetch, process, and insert the bitcoin block at `first_block_height` into burnchain DB (note that `first_block_hash` won't be needed, since the node will already have all bitcoin headers).  Then, `BurnchainDB::get_canonical_chain_tip()` will return the bitcoin block at this height as the canonical chain tip, and the `wait_for_sortitions()` method can be instructed to wait until all the block at height at `BITCOIN_TESTNET_FIRST_BLOCK_HEIGHT + burnchain.pox_constants.reward_cycle_length` has been fetched.

I don't think we need to process any prior bitcoin blocks.

Reply:
The current `next` branch is broken, and when Xenon is probably incomplete without this last step, it's a good step forward and unblocking progress on many front.
@jcnelson do you mind if we merge this as is. and address your feedback in a follow-up PR?